src/plugins/discover/public/application/main/state_management/utils/build_state_subscribe.ts

```diff
@@ -158,6 +158,17 @@ export const buildStateSubscribe =
         queryChanged: logEntry(queryChanged, prevQuery, nextQuery),
       };
 
+      if (dataState.disableNextFetchOnStateChange$.getValue()) {
```

Comment:
This file could also be much less hacky if it lived alongside the main data fetching code...